odoo/modules/registry.py

```diff
@@ -153,7 +154,11 @@ def init(self, db_name):
         self._sql_constraints = set()
         self._init = True
         self._database_translated_fields = ()  # names of translated fields in database
-        self._assertion_report = odoo.tests.result.OdooTestResult()
+        if config['test_enable']:
```

Comment:
if config['test_enable'] or config['test_file']:

Reply:
Ok, I will fix both. Ithink that there are other parts that just check one of them.